### PR TITLE
Add a yamllint job

### PR DESCRIPTION
**Pull request by vdemeester**
# Changes

Similar to #330 but for linting yamls

- Adds a yamllint image
- Adds a yamllint nightly build cron
- Adds a yamllint ci job (GitHub check, …)

/cc @afrittoli 

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

File: tekton/ci/plumbing-yamllint.yaml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+      resources:
+      - name: source
+        resource: source
+    - conditionRef: "check-name-matches"
+      params:
+      - name: gitHubCommand
+        value: $(params.gitHubCommand)
+      - name: checkName
+        value: $(params.checkName)
+  - name: build-hub
```
Comment (afrittoli) [inline tekton/ci/plumbing-yamllint.yaml:54]: I'm not sure I understand the name here?

File: tekton/ci/jobs.yaml [hunk 2]
```diff
@@ -65,3 +65,26 @@ spec:
...
+    outputs:
+      - name: endtrigger
+        type: cloudEvent
+  steps:
+  - name: lint
+    image: gcr.io/tekton-releases/dogfooding/yamllint:latest
+    script: |
+      #!/bin/sh
+      set -ex
+      yamllint .
```
Comment (afrittoli) [inline tekton/ci/jobs.yaml:90]: So, is this running on the whole repo?
Shall we have white/black list of folders? 
I don't want to yamllint vendor?

Comment (vdemeester) [reply tekton/ci/jobs.yaml:90]: yeah :stuck_out_tongue_closed_eyes: we can refine that command indeed :stuck_out_tongue_closed_eyes:

File: tekton/ci/plumbing-yamllint.yaml (hunk 1 above)
Comment (vdemeester) [reply tekton/ci/plumbing-yamllint.yaml:54]: oh crap, copy/pasta :joy_cat:

**Comment by vdemeester** [comment]
Updated :crossed_fingers:

**Pull request merged**